ccd-client/src/main/java/uk/gov/hmcts/cmc/ccd/client/service/StartCase.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.cmc.ccd.client.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.cmc.ccd.client.header.HttpHeadersFactory;
+import uk.gov.hmcts.cmc.ccd.client.model.EventRequestData;
+import uk.gov.hmcts.cmc.ccd.client.model.StartEventResponse;
+
+@Service
+public class StartCase {
+
+    private final RestTemplate restTemplate;
+    private final HttpHeadersFactory headersFactory;
+
+    @Value("${ccd.caseDataStore.baseUrl}")
+    private String ccdUrl;
+
+    @Autowired
+    public StartCase(final RestTemplate restTemplate, final HttpHeadersFactory httpHeadersFactory) {
+        this.restTemplate = restTemplate;
+        this.headersFactory = httpHeadersFactory;
+    }
+
+    public StartEventResponse exchange(final EventRequestData eventRequestData) {
+
+        final HttpEntity<String> httpEntity = new HttpEntity<>(headersFactory.getHttpHeader());
+
+        final String startCaseCreationUrl = ccdUrl + "/citizens/" + eventRequestData.getUserId()
+            + "/jurisdictions/" + eventRequestData.getJurisdictionId()
+            + "/case-types/" + eventRequestData.getCaseTypeId()
+            + "/event-triggers/" + eventRequestData.getEventId()
+            + "/token"
+            + "?ignore-warning=" + eventRequestData.getIgnoreWarning();
+
+        ResponseEntity<StartEventResponse> response = restTemplate.exchange(startCaseCreationUrl, HttpMethod.GET,
```

Comment:
this should be using feign

Reply:
We took decision not to use feign and go with proven spring rest template.

Reply:
feign is proven and works fine for simple calls only issue is with multipart upload

Reply:
it would have been same effort or less to have started with feign for this

Reply:
Overall the effort of using rest template seems fairly small I think.

Reply:
I do not buy that decision since feign is well tried as this point as well.

Reply:
As we discussed other day, we decided to leave such decision with person owning the story. Beyond ITHC, more work will be required on it and one can change it then as part of another story.

Reply:
We've already started using Feign, I don't see us rebuilding everything back to use Rest Template. If multipart upload is the only exception and the teach team is aware of it then it's fine to have one.

We have the same case in the frontend code where upload had to be handled differently with a lower level API.

Reply:
Not asking to rewrite code and start using rest template everywhere. This CCD client will only grow bigger and by the looks of it as I'm foreseeing us sending PDF as well soon due to encryption and CCD already integrated with DM. I suggest stick with REST template in this case due shortfall of Feign on multipart file (which I think is a major shortfall where we have wasted time already). Happy to look at it after ITHR as I said and we can relook at Feign if we like, just in case there's newer version of it which has fixed this problem.

Reply:
The problem is that you are approaching this from the wrong angle, the same problem in the DM approach.
We use feign for our rest calls if there is a problem that causes us to have difficultly in using it we then discuss in and perhaps have an exception.

I haven't heard of CCD supporting multipart files and as far as I know this is not under their remit and won't be having